lib/Logger/Logger.h

```diff
@@ -343,6 +338,19 @@ class Logger {
   static std::string _outputPrefix;
   static std::string _hostname;
 
-  static std::unique_ptr<LogThread> _loggingThread;
+  struct ThreadRef {
+    ThreadRef();
```

Comment:
Shall we declare this struct as non-copyable in addition? Currently it is never copied nor moved, but explicitly forbidding it feels better.

Reply:
Done